jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java

```diff
@@ -164,6 +166,8 @@ void handleDeprecatedParameters(BuildLogger logger) {
   @Parameter(defaultValue = "${project.basedir}/src/main/jib", required = true)
   private String extraDirectory;
 
+  @Nullable @Component protected SettingsDecrypter settingsDecrypter;
```

Comment:
Would this ever be `null`?

Reply:
I'm not sure.   nullaway complains without it.  I did some searching in the Maven docs but didn't find anything authoritative.

The [Github Maven Plugins](https://github.com/github/maven-plugins/blob/master/github-core/src/main/java/com/github/maven/plugins/core/GitHubProjectMojo.java#L168) look up the SettingsDecrypter through the plexus container and guard against a `ComponentLookupException`.

Reply:
Just a native question, is it possible fix encrypted passwords and plain passwords (for different servers) in `settings.xml`? If that's true, I just want to make sure `settingsDecrypter` handles plain passwords nicely. (I do believe it will.)

Reply:
I think @chanseokoh is asking about `~/.m2/settings.xml` having both encrypted and unencrypted passwords.  The default `SettingsDecryptor` passes-through unencrypted passwords.

The `SettingsDecrypter` could theoretically be replaced.  In the worse case, where the `SettingsDecrypter` is `null` but we have encrypted passwords, we will attempt to authenticate  with the encrypted password bytes.  But since it was encrypted we won't have divulged anything in the clear.  There's no solid way to identity an encrypted vs unencrypted password though.

Reply:
Mmhmm, I was mostly trying to get at the user experience side, like:

1. User sets encrypted password in Maven settings
1. We try to decrypt, but `SettingsDecrypter` is `null`, so the password wasn't decrypted
1. Users sees `401 Unauthorized` and believes their password is wrong

Whereas, I think the better UX would be:

1. User sets encrypted password in Maven settings
1. We try to decrypt, but `SettingsDecrypter` is `null`, so the password wasn't decrypted, and we give an error
1. Users sees `failed to decrypt password` (or similar)

Reply:
The default encryption approach seems to surround the passwords with braces.

We could pass in the `mavenBuildLogger` here and if `settingsDecrypter == null` and the password starts and ends with braces then issue a warning.

Reply:
In `MavenSettingsServerCredentials#retrieve` right? I might be misunderstanding how `SettingsDecrypter` works, but is there any harm in giving the warning just if `settingsDecrypter == null`?